libs/community/langchain_community/document_loaders/firecrawl.py

```diff
@@ -95,30 +217,48 @@ def __init__(
             raise ImportError(
                 "`firecrawl` package not found, please run `pip install firecrawl-py`"
             )
-        if mode not in ("crawl", "scrape"):
+        if mode not in ("crawl", "scrape", "search", "map"):
             raise ValueError(
-                f"Unrecognized mode '{mode}'. Expected one of 'crawl', 'scrape'."
+                f"Unrecognized mode '{mode}'. Expected one of 'crawl', 'scrape', 'search', 'map'."
             )
-        api_key = api_key or get_from_env("api_key", "FIRECRAWL_API_KEY")
-        self.firecrawl = FirecrawlApp(api_key=api_key, api_url=api_url)
+
+        if not url:
+            raise ValueError("Url must be provided")
+
+        api_key = api_key or get_from_env("api_key", "FIREWALL_API_KEY")
```

Comment:
Fix FIREWALL_API_KEY to FIRECRAWL_API_KEY

Reply:
done!